File: Difficulty: Medium/Max After m Range Increments/max-after-m-range-increments.cpp
class Solution{
    public:
    int findMax(int n, vector<int> &a, vector<int> &b, vector<int> &k){
        int m=a.size();
        
        map<int, int> mp;
        for(int i=0; i<m; i++){
            mp[a[i]] += k[i];
            mp[b[i]+1] -= k[i];
        }
        
        int sum=0;
        int res=0;
        for(auto it: mp){
            sum += it.second;
            res=max(res, sum);
        }
        
        return res;
        
    }
};

File: Difficulty: Medium/Max After m Range Increments/max-after-m-range-increments.py
class Solution:
    def findMax(self, n, a, b, k):
        m=len(a)
        
        mp={}
        for i in range(m):
            mp[a[i]] = mp.get(a[i], 0)+k[i]
            mp[b[i]+1] = mp.get(b[i]+1, 0)-k[i]
        
        sum=0
        res=0
        for key in sorted(mp):
            sum += mp[key]
            res=max(res, sum)
        
        return res
        
        
        